File: erp-frontend/src/types/product.ts
export interface Product {
  id: number;
  name: string;
  description?: string;
  category?: ProductCategory;
  unit?: Unit;
  sku?: string;
  barcode?: string;
  active: boolean;
  manufacturer?: string;
  hsnCode?: string;
  taxRate: number;
  price?: number;
  minStockLevel: number;
  maxStockLevel?: number;
  costPrice?: number;
  prices?: ProductPrice[];
  createdAt?: string;
  updatedAt?: string;
}

export interface ProductCategory {
  id: number;
  name: string;
  description?: string;
  parentCategory?: ProductCategory;
}

export interface Unit {
  id: number;
  name: string;
  symbol?: string;
}

export interface ProductPrice {
  id: number;
  product?: Product;
  price: number;
  effectiveFrom?: string;
  effectiveTo?: string;
}

File: src/main/java/com/erp/system/product/entity/Product.java
package com.erp.system.product.entity;

import com.erp.system.common.entity.BaseAuditEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "product")
@Getter
@Setter
public class Product extends BaseAuditEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private ProductCategory category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "unit_id")
    private Unit unit;

    @Column(unique = true)
    private String sku;

    private String barcode;

    private Boolean active = true;

    private String manufacturer;
    private String hsnCode;

    @Column(precision = 5, scale = 2)
    private BigDecimal taxRate = BigDecimal.ZERO;

    private Integer minStockLevel = 0;
    private Integer maxStockLevel;

    @Column(precision = 10, scale = 2)
    private BigDecimal costPrice;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<ProductPrice> prices = new ArrayList<>();

    public void setPrice(@NotNull(message = "Price is required") @DecimalMin(value = "0.01", message = "Price must be greater than 0") BigDecimal price) {
    }
}